Sources/CasePaths/EnumReflection.swift

```diff
@@ -67,89 +67,48 @@ public func extract<Root, Value>(case embed: @escaping (Value) -> Root, from roo
 /// - Parameter embed: An enum case initializer.
 /// - Returns: A function that can attempt to extract associated values from an enum.
 public func extract<Root, Value>(_ embed: @escaping (Value) -> Root) -> (Root) -> (Value?) {
-  guard let metadata = EnumMetadata(Root.self) else {
-    #if DEBUG
-      print(
-        "\(#function) can never extract values from \(Root.self) because \(Root.self) isn't an enum!"
-      )
-    #endif
-    return { _ in nil }
-  }
+  guard
+    let metadata = EnumMetadata(Root.self),
+    metadata.typeDescriptor.fieldDescriptor != nil
+  else { return { _ in nil } }
 
-  guard metadata.typeDescriptor.fieldDescriptor != nil else {
-    #if DEBUG
-      print(
-        "\(#function) can never extract values from \(Root.self) because the metadata for \(Root.self) is incomplete!"
-      )
-    #endif
-    return { _ in nil }
-  }
+  var cachedTag: UInt32?
+  var cachedStrategy: Strategy<Root, Value>?
 
-  var cachedTag: UInt32? = nil
-  var cachedStrategy: Strategy<Root, Value> = .unimplemented
   return { root in
     let metadata = EnumMetadata(assumingEnum: Root.self)
     let rootTag = metadata.tag(of: root)
 
-    if let cachedTag = cachedTag {
+    if let cachedTag = cachedTag, let cachedStrategy = cachedStrategy {
       guard rootTag == cachedTag else { return nil }
       return cachedStrategy.extract(from: root, tag: rootTag)
     }
 
-    guard
-      let rootStrategy = Strategy<Root, Value>(tag: rootTag),
-      let value = rootStrategy.extract(from: root, tag: rootTag)
-    else {
-      return nil
-    }
+    guard let value = Strategy<Root, Value>(tag: rootTag).extract(from: root, tag: rootTag)
+    else { return nil }
 
-    let embeddedValue = embed(value)
-    let embedTag = metadata.tag(of: embeddedValue)
+    let embedTag = metadata.tag(of: embed(value))
     cachedTag = embedTag
+    cachedStrategy = Strategy<Root, Value>(tag: embedTag)
 
-    guard cachedTag == rootTag else {
-      // I have the correct tag now, so I can now pick the correct strategy, even if I won't use it this time.
-      if let embedStrategy = Strategy<Root, Value>(tag: embedTag) {
-        cachedStrategy = embedStrategy
-      } else {
-        #if DEBUG
-          print(
-            "\(#function) can never extract values from a \(Root.self) case like \(embeddedValue) because I don't know how to convert its associated value to \(Value.self)!"
-          )
-        #endif
-        cachedStrategy = .unimplemented
-      }
+    guard cachedTag == rootTag else { return nil }
```

Comment:
This should be `embedTag == rootTag`. And it was, before I started working on the code. I messed it up.